service/history/api/updateworkflow/api.go

```diff
@@ -216,6 +218,7 @@ func (u *Updater) ApplyRequest(
 		ms.GetMostRecentWorkerVersionStamp(),
 		ms.HasCompletedAnyWorkflowTask(),
 	)
+	u.workflowVersioningInfo = workflow.GetWorkflowVersioningInfoMatchingTask(ms)
```

Comment:
```suggestion
	u.workflowVersioningInfo = common.CloneProto(workflow.GetWorkflowVersioningInfoMatchingTask(ms))
```

Reply:
and `u.directive` above too.

Reply:
workflowVersioningInfo is removed and u.directive does not come from MS, it's created fresh.